specification/scvmm/ScVmm.Management/models.tsp

```diff
@@ -640,6 +643,36 @@ model OsProfileForVmInstance {
   /** Gets os version. */
   @visibility("read")
   osVersion?: string;
+
+  /** Gets or sets the domain name. */
+  @added(Versions.v2024_06_01)
+  domainName?: string;
+
+  /** Gets or sets the domain username. */
+  @added(Versions.v2024_06_01)
+  domainUsername?: string;
+
+  /** Password of the domain the VM has to join. */
+  @visibility("create", "update")
+  @secret
+  @added(Versions.v2024_06_01)
+  domainPassword?: string;
+
+  /** Gets or sets the workgroup. */
+  @added(Versions.v2024_06_01)
+  workgroup?: string;
+
+  /** Gets or sets the product key.Input format xxxxx-xxxxx-xxxxx-xxxxx-xxxxx */
+  @added(Versions.v2024_06_01)
+  productKey?: string;
```

Comment:
>productKey [](http://example.com/codeflow?start=2&length=10)

does it quality for a secret? you want to return the product-key it in get/read calls?

Reply:
Have made it a secret.